src/libraries/System.Net.Http/src/System/Net/Http/BrowserHttpHandler/BrowserHttpHandler.cs

```diff
@@ -0,0 +1,534 @@
+                await stream.WriteAsync(data, 0, data.Length, cancellationToken).ConfigureAwait(continueOnCapturedContext: true);
+            }
+            protected internal override bool TryComputeLength(out long length)
+            {
+                if (_data != null)
+                {
+                    length = _data.Length;
+                    return true;
+                }
+
+                length = 0;
+                return false;
+            }
+
+            protected override void Dispose(bool disposing)
+            {
+                _status?.Dispose();
+                base.Dispose(disposing);
+            }
+        }
+
+        private sealed class WasmHttpReadStream : Stream
+        {
+            private WasmFetchResponse? _status;
+            private JSObject? _reader;
+
+            private byte[]? _bufferedBytes;
+            private int _position;
+
+            public WasmHttpReadStream(WasmFetchResponse status)
+            {
+                _status = status;
+            }
+
+            public override bool CanRead => true;
+            public override bool CanSeek => false;
+            public override bool CanWrite => false;
+            public override long Length => throw new NotSupportedException();
+            public override long Position
+            {
+                get => throw new NotSupportedException();
+                set => throw new NotSupportedException();
+            }
+
+            public override async Task<int> ReadAsync(byte[] buffer, int offset, int count, CancellationToken cancellationToken)
+            {
+                if (buffer == null)
+                {
+                    throw new ArgumentNullException(nameof(buffer));
+                }
+                if (offset < 0)
+                {
+                    throw new ArgumentOutOfRangeException(nameof(offset));
+                }
+                if (count < 0 || buffer.Length - offset < count)
+                {
+                    throw new ArgumentOutOfRangeException(nameof(count));
+                }
+
+                if (_reader == null)
+                {
+                    // If we've read everything, then _reader and _status will be null
+                    if (_status == null)
+                    {
+                        return 0;
+                    }
+
+                    try
+                    {
+                        using (JSObject body = _status.Body)
+                        {
+                            _reader = (JSObject)body.Invoke("getReader");
+                        }
+                    }
+                    catch (JSException)
+                    {
+                        cancellationToken.ThrowIfCancellationRequested();
+                        throw;
+                    }
+                }
+
+                if (_bufferedBytes != null && _position < _bufferedBytes.Length)
```

Comment:
Am I reading this correctly that if someone reads to the end but then calls read again (such that _position >= _bufferedBytes.Length), they'll end up starting to read again at the beginning because they'll fail this if check and proceed to the below try?  If so, that seems wrong.